src/System.Drawing.Common/tests/IconTests.cs

```diff
@@ -240,13 +240,41 @@ public void Ctor_NullIcon_ThrowsArgumentNullException()
         }
 
         [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
-        [ActiveIssue(21360)]
-        public void Ctor_InvalidIconHandle_SetsHandleToZero()
+        public void Ctor_InvalidHandle_Success()
         {
-            using (Icon source = Icon.FromHandle((IntPtr)100))
-            using (var icon = new Icon(source, 10, 10))
+            var random = new Random();
+
+            // First, generate a random handle. There are 2^32 numbers in the range, so it's likely that
+            // there will be no such icon associated with this handle.
+            // However, if there is an icon associated with this handle, the test will fail.
+            // Instead, repeat 100 times with another random number. The chance of an icon existing 100 times
+            // with a random handle is minimal.
+            for (int i = 0; i < 100; i++)
             {
-                Assert.Throws<ObjectDisposedException>(() => icon.Handle);
+                IntPtr handle = (IntPtr)random.Next();
+
+                try
+                {
+                    using (Icon icon = Icon.FromHandle(handle))
```

Comment:
If it does happen to conflict with an existing handle, is this going to then potentially close that other handle causing problems for whatever code is using the other one?